pages/api/modifyuser.js

```diff
@@ -47,6 +61,21 @@ export default async function handle(req, res) {
     body.role = undefined;
   }
 
+  //If user attempts to change a role from Admin to non-Admin, then the role change will be denied.
+  //The user is still able to change both the Name and Email.
+  if (userToBeChanged.role === 'ADMIN' && body.role !== 'ADMIN') {
+    await prisma.user.update({
+      where: {
+        id: body.id
+      },
+      data: {
+        name: body.name,
+        email: body.email
+      }
+    });
+    //202 return indicates the request is accepted, but not entirely completed.
+    return res.status(202).end();
```

Comment:
My opinion is we should just block the request if it's bad. Especially since this response has no body explaining what parts of the request did/didn't went though.

Reply:
This version allows users to make changes to the name and email, but not the role if the role is changing from Admin to non-Admin. Do you suggest blocking all changes whenever a user is modifying an ADMIN's role? If so, in the next PR, shall we also make a pop-up message to explain why this request is denied? (Ex: DO NOT MODIFY ADMIN'S ROLE)

Reply:
In general you should always block bad requests so callers know to stop making them. Also, we return a 202 without giving any response body, if we send this then callers will not know what actually got processed and what didn't.

I suggest we block the request from the API side (this API is exposed to the public internet), and then on the UI we prevent any bad requests to be possible. Ideally roles will be a dropdown menu, and for a user that is an ADMIN we just don't let them change the role, don't even give them the option to.

Reply:
Got it! Thank you, Robert! We will edit the code in this pull request to completely block the bad requests, and return 403 (Forbidden) if it happens. And we will also try to edit the UI to ensure the users won't be able to submit bad requests.